lib/src/gutter_padding.dart

```diff
@@ -0,0 +1,98 @@
+import 'package:flutter/widgets.dart';
+
+import '../flutter_gutter.dart';
+
+/// A widget that insets its child by the material padding or the given padding.
+class GutterPadding extends Gutter {
+  /// Creates a new [GutterPadding] widget with custom padding.
+  const GutterPadding({
+    super.key,
+    this.child,
+    super.size,
+    super.type = GutterType.medium,
+    super.scaleFactor = 1,
+  })  : left = null,
+        top = null,
+        right = null,
+        bottom = null;
+
+  /// Creates a new [GutterPadding] widget with specific edge insets.
+  const GutterPadding.only({
+    super.key,
+    this.child,
+    this.left,
+    this.top,
+    this.right,
+    this.bottom,
+    super.type = GutterType.medium,
+    super.scaleFactor = 1,
+  });
+
+  /// Creates a new [GutterPadding] widget with symmetric padding.
+  const GutterPadding.symmetric({
+    super.key,
+    this.child,
+    Gutter? vertical,
+    Gutter? horizontal,
+    super.type = GutterType.medium,
+    super.scaleFactor = 1,
+  })  : left = horizontal,
+        top = vertical,
+        right = horizontal,
+        bottom = vertical;
+
+  /// Creates a new [GutterPadding] widget with uniform padding.
+  const GutterPadding.all({
+    super.key,
+    Gutter? value,
+    this.child,
+    super.type = GutterType.medium,
+    super.scaleFactor = 1,
+  })  : left = value,
+        top = value,
+        right = value,
+        bottom = value;
+
+  /// The child widget.
+  final Widget? child;
+
+  /// The offset from the left.
+  final Gutter? left;
```

Comment:
It feels really weird to me to pass Widgets in to configure padding sizes. I think this  feature (a gutter padding) would be VERY useful, but I think the API is hard to get right.

It's pretty tedious to write, but I think the best API might be:
`GutterPaddingLarge.only(<bools for each direction>)`
`GutterPaddingLarge.symmetric(<bools>)`
`GutterPaddingLarge.all()`
and repeated for each size Gutter. But I don't love this either, it's still a bit clunky and I wonder if it's that much better than just using the built in padding and edge inset system with `context.gutter` etc...

All in all I lean towards not merging this now until we come up with an API that feels clean. Happy to soundboard with you one what the API should look like, and if we do find something that feels good I don't mind doing the implementation myself if you don't feel like spinning up more PRs.

Can you snip this widget from the PR and I'll merge and ship?

Reply:
I used a double before but that is not great either because the default should be to calculate the right gutter size when not hardcode to anything. This worked great for me but i agree it should rather not be a Widget itself. I'll remove it out for now and work in a new PR on it.

Reply:
#7